os/arch/arm/src/stm32l4/stm32l4_idle.c

```diff
@@ -67,6 +75,70 @@
 #endif
 
 #define PM_IDLE_DOMAIN 0 /* Revisit */
+#define CONFIG_TICKLESS_TIME_CORRECTION
+#define STM32L4_EXT_JOYKEY_ACTIVITY			4
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void up_extiisr(int irq, uint32_t *regs)
+{
+	HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_13);
+	up_lowputc('i');
+	pm_activity(PM_IDLE_DOMAIN, STM32L4_EXT_JOYKEY_ACTIVITY);
+}
+
+void set_exti_button(void)
+{
+	GPIO_InitTypeDef GPIO_InitStruct = {0};         /* GPIO Ports Clock Enable */
+	__HAL_RCC_GPIOA_CLK_ENABLE();
+	__HAL_RCC_GPIOA_CLK_SLEEP_ENABLE();
+	__HAL_RCC_GPIOC_CLK_ENABLE();   /*Configure GPIO pin : PC13 */
+	__HAL_RCC_GPIOC_CLK_SLEEP_ENABLE();
+	GPIO_InitStruct.Pin = GPIO_PIN_13;
+	GPIO_InitStruct.Mode = GPIO_MODE_IT_RISING;
+	GPIO_InitStruct.Pull = GPIO_NOPULL;
+	HAL_GPIO_Init(GPIOC, &GPIO_InitStruct); /* EXTI interrupt init*/
+	HAL_NVIC_SetPriority(EXTI15_10_IRQn, 2, 0);
+	HAL_NVIC_EnableIRQ(EXTI15_10_IRQn);
+	(void)irq_attach(STM32L4_IRQ_EXTI1510, (xcpt_t)up_extiisr, NULL);
+}
+
+/****************************************************************************
+ * Name: pm_sleep
+ *
+ * Description:
+ *   Perform state transition from NORMAL to SLEEP
+ *
+ ****************************************************************************/
+int pm_sleep(void)
+{
+        int ret;
+        irqstate_t flags;
+
+        flags = irqsave();
+        sched_lock();
+        ret = pm_changestate(0, PM_SLEEP);
+        if (ret < 0)
+        {
+                printf("change state failed\n");
+                /* The new state change failed, revert to the preceding state */
+                (void)pm_changestate(0, PM_NORMAL);
+                return ERROR;
```

Comment:
I think you'd better restore the irq before return.

Reply:
Agree. Will modify